src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -125,25 +133,14 @@ public bool CanNavigateToPosition(Workspace workspace, DocumentId documentId, in
 
         public bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, OptionSet options)
         {
-            // Navigation should not change the context of linked files and Shared Projects.
-            documentId = workspace.GetDocumentIdInCurrentContext(documentId);
```

Comment:
All the logic in this method was duplicated across multiple, I just combined them all